pkg/issuer/acme/setup.go

```diff
@@ -166,6 +166,11 @@ func (a *Acme) Setup(ctx context.Context) error {
 
 	// registerAccount will also verify the account exists if it already
 	// exists.
+	if a.issuer.GetStatus().ACME.LastRegisteredEmail == a.issuer.GetSpec().ACME.Email {
```

Comment:
I'm unsure whether we want to explicitly bail out/stop processing here *just* if the status == the spec (i.e. what if the users private key was deleted? In this instance we should be generating a new private key and re-registering an account).

Also, the proper way to 'handle' updating the email field is for us to actually update the email address associated with the ACME account. You can see the actual method you can use to do this here: https://github.com/jetstack/cert-manager/blob/9c6d81a92a192eb95be90084956dc04c0cb7552c/third_party/crypto/acme/acme.go#L371-L373

In terms of how the issuer's `status.registeredEmail` field is set, I think to make things clearer and easier to keep consistent, this field should *always* be set to the value on the `Account` object returned from the ACME server, instead of setting it to the value of `spec.acme.email` (as it could fall out of sync and we could be incorrectly reporting the status).

Reply:
I'll have another look, thanks!